### PR TITLE
Pbi 27770

### PR DESCRIPTION
**Pull request by jonnolen**
@NullVoxPopuli

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/data/query.tsx [hunk 1]
```diff
@@ -136,7 +136,8 @@ export function queryApi<T>(mapRecordsToProps, options?: IQueryOptions) {
         const dataProps = {
           ...result,
           error,
-          isLoading
+          isLoading,
+          ackbar: this.tryFetch.bind(this, true),
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/data/query.tsx]: ackbar? :)
we could just make `refetch` a thing :)

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { compose } from 'recompose';
+import { withData, WithDataProps } from 'react-orbitjs';
+import { withTranslations } from '@lib/i18n';
+import UserInput from './user-input';
+
+interface IOwnProps{
+  onUserAdded: () => Promise<void>;
+}
+
+export type IProps = IOwnProps
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/index.tsx]: do you need to export this type?

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/user-input.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import { IAttributeProps } from '@lib/dom';
+
+export interface IProps {
+  onSubmit: (email: string) => void;
+  placeholder?: string;
+}
+
+export interface IState {
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/user-input.tsx]: Do we need to export the state interface and IProps?

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/index.tsx (hunk 2 above)
Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/index.tsx]: it doesn't do any harm to export it, as types aren't included in the transpiled code

File: source/SIL.AppBuilder.Portal.Frontend/src/translations/locales/es-419.json [hunk 4]
```diff
@@ -278,6 +278,12 @@
   "users": {
     "noneFound": "No se encontraron usuarios con esos criterios de búsqueda.",
     "title": "Administrar usuarios",
+    "addUser": {
```
Comment (giancorzo) [inline source/SIL.AppBuilder.Portal.Frontend/src/translations/locales/es-419.json]: that is not Spanish!

      "placeholder": "Correo electrónico",
      "button": "Agregar Usuario", //typo
      "modalTitle": "Nuevo usuario",
      "modalAddButton": "Agregar"

In any case, we don't need to add locale in different languages because they will get override by Crowind push

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/translations/locales/es-419.json]: crowdin requires all non-english translations be handled in crowdin. 
I don't know what the crowdin URL is for this project, but here is what one looks like for https://emberclear.io : https://crowdin.com/project/emberclear

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/user-input.tsx (hunk 3 above)
Comment (giancorzo) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/add/user-input.tsx]: it doesn't but confuses the vscode and you see 100 IProvidedProps as recommendations :(

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/debounced-search-field/index.tsx [hunk 5]
```diff
@@ -31,13 +31,13 @@ export default class DebouncedSearch extends React.Component<IProps & IAttribute
   }
 
   render() {
-    const { placeholder, className, ...other } = this.props;
+    const { placeholder, className, onSubmit, ...other } = this.props;
     const { searchTerm } = this.state;
 
-    const attributes = filterForValidAttributes(other);
+    // const attributes = filterForValidAttributes(other);
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/debounced-search-field/index.tsx]: comment?

Comment (jonnolen) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/debounced-search-field/index.tsx]: good catch.

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/products/index.tsx [hunk 6]
```diff
@@ -93,7 +93,6 @@ class Products extends React.Component<IProps> {
           {t('project.products.title')}
         </h3>
         {
-          !isEmpty(products) &&
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/products/index.tsx]: why was this removed?

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/__tests__/add-user-acceptance-test.tsx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+        beforeEach( async () => {
+          await when(() => page.addUserModal.isVisible);
+        });
+
+        describe("when user exists in system", () => {
+          const existingEmail = "existing@foo.com";
+          beforeEach( async function() {
+            let newUsersData = {...usersData};
+            let newUser = {...user};
+            newUser.id = '2';
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/users/list/__tests__/add-user-acceptance-test.tsx]: this is a neat way to reduce duplication between tests :+1:

**Comment by NullVoxPopuli** [comment]
@jonnolen can you provide a couple screenshots?

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/products/index.tsx (hunk 6 above)
Comment (jonnolen) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/products/index.tsx]: maybe i lied.

Comment (jonnolen) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/products/index.tsx]: another good/lucky catch.

**Comment by giancorzo** [comment]
The popup doesn't follow the app style.
"Add button" should be in the middle bottom and the input doesn't follow the style either (check other forms). 

Also, perhaps the "+new" button should be in the next line to follow organization settings style.
And do we need a popup if we only have email input? even with name and last name inputs, this could be an inline form on the user page? what do you think?

**Comment by NullVoxPopuli** [comment]
@giancorzo I'd say we make a bug for later, and move forward. Lots to do :)
